hazelcast/src/main/java/com/hazelcast/spi/impl/eventservice/impl/EventServiceImpl.java

```diff
@@ -441,16 +492,10 @@ public void executeEventCallback(Runnable callback) {
 
     @Override
     public void handle(Packet packet) {
-        try {
-            eventExecutor.execute(new RemoteEventProcessor(this, packet));
-        } catch (RejectedExecutionException e) {
-            rejectedCount.inc();
-
-            if (eventExecutor.isLive()) {
-                Connection conn = packet.getConn();
-                String endpoint = conn.getEndPoint() != null ? conn.getEndPoint().toString() : conn.toString();
-                logFailure("EventQueue overloaded! Failed to process event packet sent from: %s", endpoint);
-            }
+        EventEnvelope eventEnvelope = (EventEnvelope) nodeEngine.toObject(packet);
```

Comment:
@ahmetmircik,

there is a problem with this approach. I can understand - is message should be processed on dedicated thread or on any event service thread - after message de-serialization only:

`

        EventEnvelope eventEnvelope = (EventEnvelope) nodeEngine.toObject(packet);
        Registration registration = getRegistration(eventEnvelope);
        if (registration.isMultithreaded()) {
            // process it via regular Runable task
        } else {
            // process it via StripedRunable task
        }
`

what would you suggest?